crates/shared/src/price_estimation/trade_finder.rs

```diff
@@ -80,24 +84,49 @@ impl TradeEstimator {
     }
 
     async fn estimate(&self, query: Query) -> Result<Estimate, PriceEstimationError> {
+        let owner = query.from.unwrap_or_default();
```

Comment:
Should we just error out early? Personally I don't think its worth doing a proper trade verification if `query.from` is `None` as it will almost certainly fail, and uses valuable API requests to DEX aggregator APIs and our node.

Reply:
Now I remember why I did it this way. If we error out some unit tests are complaining and I didn't want to blow up the diff even further because the PR in itself is already broken.
Will continue to simply `warn` for now be leave the PR unmerged to see if the remaining changes are simply enough to be merged together with this PR to not temporarily break verification.